frontend/src/metabase/visualizations/visualizations/SmartScalar.jsx

```diff
@@ -0,0 +1,192 @@
+  _scalar: ?HTMLElement;
+
+  static settings = {
+    ...columnSettings({
+      getColumns: ([{ data: { cols } }], settings) => [
+        _.find(cols, col => col.name === settings["scalar.field"]) || cols[1],
+      ],
+    }),
+    "scalar.switch_positive_negative": {
+      title: t`Switch positive / negative colors?`,
+      widget: "toggle",
+    },
+  };
+
+  // Always return true here so that we don't discourage this feature.
+  static isSensible() {
+    return true;
+  }
+
+  // Smart scalars need to have a breakout
+  static checkRenderable(series, settings) {
+    const singleSeriesHasNoRows = ({ data: { cols, rows } }) => rows.length < 1;
+    if (_.every(series, singleSeriesHasNoRows)) {
+      throw new MinRowsError(1, 0);
+    }
+
+    const cols = series[0].data.cols;
+
+    if (cols.length < 2) {
+      throw new NoBreakoutError(
+        t`Group by a time field to see how this has changed over time`,
+      );
+    }
+  }
+
+  _getColumnIndex(cols: Column[], settings: VisualizationSettings) {
+    const columnIndex = _.findIndex(
+      cols,
+      col => col.name === settings["scalar.field"],
+    );
+    return columnIndex < 0 ? 1 : columnIndex;
+  }
+
+  render() {
+    const insights =
+      this.props.rawSeries &&
+      this.props.rawSeries[0].data &&
+      this.props.rawSeries[0].data.insights;
+    const {
+      onChangeCardAndRun,
+      onVisualizationClick,
+      isDashboard,
+      isFullscreen,
+      settings,
+      visualizationIsClickable,
+      series: [{ card, data: { rows, cols } }],
+    } = this.props;
+
+    let granularity;
+    if (Card.isStructured(card)) {
+      const query = Card.getQuery(card);
+      const breakouts = query && Query.getBreakouts(query);
+      granularity = formatBucketing(
+        parseFieldBucketing(breakouts[0]),
+      ).toLowerCase();
+    }
+
+    const change = formatNumber(insights["last-change"] * 100);
+    const isNegative = (change && Math.sign(change) < 0) || false;
+
+    let color = isNegative ? colors["error"] : colors["success"];
+
+    // if the number is negative but thats been identified as a good thing (e.g. decreased latency somehow?)
+    if (isNegative && settings["scalar.switch_positive_negative"]) {
+      color = colors["success"];
+    } else if (!isNegative && settings["scalar.switch_positive_negative"]) {
+      color = colors["error"];
+    }
+
+    const changeDisplay = (
+      <span style={{ fontWeight: 900 }}>{Math.abs(change)}%</span>
+    );
+    const separator = (
+      <span
+        style={{
+          color: colors["text-light"],
+          fontSize: "0.6rem",
+          marginLeft: 4,
+          marginRight: 4,
+        }}
+      >
+        •
+      </span>
+    );
+    const granularityDisplay = <span>{jt`last ${granularity}`}</span>;
+
+    const columnIndex = this._getColumnIndex(cols, settings);
+    const value = rows[0] && rows[0][columnIndex];
+    const column = cols[columnIndex];
+    const clicked = { value, column };
```

Comment:
To get the drill through behavior @mazameli described you need to include the `dimensions` array, e.x. https://github.com/metabase/metabase/blob/master/frontend/src/metabase/visualizations/components/FunnelNormal.jsx#L117-L126

As mentioned above unless the metric and dimension are configurable you could probably just hardcode dimensionIndex to 0 and metricIndex to 1.

Also, I don't think it matters much since we don't actually use `value` in drill throughs yet, but you're using the first value rather than last value. I think `last-value` or `rows[rows.length - 1][metricIndex]` would be more correct.